<launch>
    <arg name="map" default="simple"/>

    <!--    Publish map-->
    <node pkg="map_server" name="map_server" type="map_server"
          args="$(find full_coverage_path_planner)/maps/$(arg map).yaml"
          respawn="true" clear_params="true" output="screen"/>

    <!--    Computes number of agents required from map-->
    <include file="$(find full_coverage_path_planner)/launch/optimizer.launch"/>

    <!--    Compute coverage path from map, divide path for each agent -->
    <include file="$(find full_coverage_path_planner)/launch/path_planner.launch"/>

    <!--    Waits for number of agents and waypoints for each agent, then launches the entire stack for each agent-->
    <node pkg="full_coverage_path_planner" type="multi_robot_bringup.py" name="multi_robot_bringup" clear_params="true"
          required="true"
          output="screen">
        <param name="robot_namespace" value="robot"/>
    </node>

    <node pkg="multirobot_map_merge" type="map_merge" respawn="false" name="map_merge" clear_params="true">
        <param name="robot_map_topic" value="coverage_grid"/>
        <param name="robot_namespace" value="robot"/>
        <param name="merged_map_topic" value="coverage_grid_merged"/>
        <param name="world_frame" value="map"/>
        <param name="known_init_poses" value="true"/>
        <param name="merging_rate" value="4.0"/>
        <param name="discovery_rate" value="0.05"/>
    </node>

        <node pkg="rviz" type="rviz" name="rviz"
              args="-d $(find full_coverage_path_planner)/rviz/rviz.rviz"/>

</launch>